components/ui/Dialog.tsx

```diff
@@ -0,0 +1,250 @@
+    close:
+      'focus:ring-ring absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-accent data-[state=open]:text-muted-foreground',
+    title: 'flex items-center gap-1.5 text-xl font-bold',
+    contentMain: 'text-foreground',
+    footer: 'flex flex-col-reverse sm:flex-row sm:justify-end gap-2',
+  },
+  variants: {
+    type: {
+      [DialogVariants.Error]: {
+        contentMain: 'text-sm',
+      },
+      [DialogVariants.Prompt]: {
+        // contentMain: '',
+      },
+      [DialogVariants.Confirm]: {
+        // contentMain: '',
+      },
+      [DialogVariants.Info]: {
+        // contentMain: '',
+      },
+    },
+  },
+});
+
+// Main Dialog Component
+type DialogProps = {
+  dialog: DialogType;
+  dialogOrder: number;
+  handleOpenChange: (dialog: DialogType) => void;
+  confirmDialog: (dialog: DialogType) => void;
+  cancelDialog: (dialog: DialogType) => void;
+};
+
+const Dialog = (props: DialogProps) => {
+  const { dialog, handleOpenChange, cancelDialog, confirmDialog, dialogOrder } =
+    props;
+
+  const { title, contentMain, overlay, content, footer, close } =
+    dialogElements({
+      type: dialog.type,
+    });
+
+  return (
+    <DialogPrimitive.Root open onOpenChange={() => handleOpenChange(dialog)}>
+      <DialogPrimitive.Portal>
+        <DialogPrimitive.Overlay className={overlay()}>
+          <DialogContent className={content()} dialogOrder={dialogOrder}>
+            <DialogPrimitive.Title className={title()}>
+              <DialogIcon variant={dialog.type} />
+              <span>{dialog.title}</span>
+            </DialogPrimitive.Title>
+            <main className={contentMain()}>
+              {dialog.content}
+              <br />
+              {dialog.type === 'Error' && <DialogError {...dialog.error} />}
+            </main>
+            <DialogFooter
+              className={footer()}
+              dialog={dialog}
+              cancelDialog={cancelDialog}
+              confirmDialog={confirmDialog}
+            />
+            <DialogClose
+              className={close()}
+              disable={dialog.type === 'Prompt'}
+            />
+          </DialogContent>
+        </DialogPrimitive.Overlay>
+      </DialogPrimitive.Portal>
+    </DialogPrimitive.Root>
+  );
+};
+export default Dialog;
+
+// Dialog Content Section
+type DialogContentProps = {
+  dialogOrder: number;
+} & React.ComponentPropsWithoutRef<typeof DialogPrimitive.Content>;
+
+const DialogContent = React.forwardRef<
+  React.ElementRef<typeof DialogPrimitive.Content>,
+  DialogContentProps
+>(({ children, dialogOrder, ...props }, ref) => {
+  const variants = React.useMemo(
+    () => ({
+      visible: {
+        opacity: 1,
+        translateY: '-50%',
+        translateX: '-50%',
+        scale: 1,
+        transition: { delay: dialogOrder * 0.25 },
```

Comment:
This should use stagger children on the parent rather than implementing a delay here. https://www.framer.com/motion/transition/#%23%23staggerchildren

Reply:
I tried to use the `staggerChildren` feature, but couldn't make it work. Here's an explanation about my attempts :https://cshd.slack.com/archives/C054Y8B8308/p1719218516835389